zerver/tests/test_subs.py

```diff
@@ -816,6 +816,34 @@ def test_successful_subscriptions_add(self):
                 add_streams, self.streams, self.test_email, self.streams + add_streams)
         self.assert_length(events, 4, True)
 
+    def test_successful_subscriptions_add_with_announce(self):
+        # type: () -> None
+        """
+        Calling POST /json/users/me/subscriptions should successfully add
+        streams, and should determine which are new subscriptions vs
+        which were already subscribed. We add 2 new streams to the
+        list of subscriptions and confirm the right number of events
+        are generated.
+        """
+        self.assertNotEqual(len(self.streams), 0)  # necessary for full test coverage
+        add_streams = [u"Verona2", u"Denmark5"]
+        self.assertNotEqual(len(add_streams), 0)  # necessary for full test coverage
+        events = [] # type: List[Dict[str, Any]]
+        other_params = {
+            'announce': 'true',
+        }
+        notifications_stream = Stream.objects.get(name=self.streams[0], realm=self.realm)
+        self.realm.notifications_stream = notifications_stream
+        self.realm.save()
+
+        # Delete the UserProfile from the cache so the realm change will be
+        # picked up
+        cache.cache_delete(cache.user_profile_by_email_cache_key(self.test_email))
+        with tornado_redirected_to_list(events):
+            self.helper_check_subs_before_and_after_add(self.streams + add_streams, other_params,
+                add_streams, self.streams, self.test_email, self.streams + add_streams)
+        self.assert_length(events, 7, True)
```

Comment:
I think you meant to say `assertEqual` there?  Maybe not.  At any rate, see my note elsewhere in this ticket's conversation about using a `schema` against events.  It may be way overkill for this series, so it's just food for thought.